common/common.go

```diff
@@ -94,77 +96,61 @@ func getBlockFromRPC(rpcClient *rpcclient.Client, height int) (*walletrpc.Compac
 	return block.ToCompact(), nil
 }
 
-func BlockIngestor(rpcClient *rpcclient.Client, cache *BlockCache, log *logrus.Entry,
-	stopChan chan bool, startHeight int) {
+func BlockIngestor(rpcClient *rpcclient.Client, cache *BlockCache, log *logrus.Entry, startHeight int) {
 	reorgCount := 0
 	height := startHeight
-	timeoutCount := 0
 
 	// Start listening for new blocks
+	retryCount := 0
 	for {
-		select {
-		case <-stopChan:
-			break
-
-		case <-time.After(15 * time.Second):
-			for {
-				if reorgCount > 0 {
-					height -= 10
-				}
-
-				if reorgCount > 10 {
-					log.Error("Reorg exceeded max of 100 blocks! Help!")
-					return
-				}
-
-				block, err := getBlockFromRPC(rpcClient, height)
-
-				if err != nil {
+		block, err := getBlockFromRPC(rpcClient, height)
+		if block == nil || err != nil {
+			if err != nil {
+				log.WithFields(logrus.Fields{
+					"height": height,
+					"error":  err,
+				}).Warn("error with getblock rpc")
+				retryCount++
+				if retryCount > 10 {
 					log.WithFields(logrus.Fields{
-						"height": height,
-						"error":  err,
-					}).Warn("error with getblock")
-
-					timeoutCount++
-					if timeoutCount == 3 {
-						log.WithFields(logrus.Fields{
-							"timeouts": timeoutCount,
-						}).Warn("unable to issue RPC call to zcashd node 3 times")
-						break
-					}
+						"timeouts": retryCount,
+					}).Fatal("unable to issue RPC call to zcashd node")
 				}
+			}
+			// We're up to date in our polling; wait for a new block
+			time.Sleep(10 * time.Second)
```

Comment:
This entire function should better handle `zcashd` disappearing or not responding for any reason.